### ISSUE TITLE
kubeflow 1.1.0 install failed when use 1.18 or higher kubernetes

### ISSUE THREAD

**Issue by whalecold** [kubeflow/manifests#1543]
The seldon CRD [APIV3Schema](https://github.com/kubeflow/manifests/blob/ed615418ef15e54c9a68979d8d2fc45e17bab7b1/seldon/seldon-core-operator/base/resources.yaml#L2490) is invalid in the kubernetes of 1.18，Could we set the `protocol`  field to `Required` ? Alternative, using the v1 CRD API.

Ref: https://github.com/kubernetes/kubernetes/issues/91395#issuecomment-633261523

**Comment by tomalbrecht** [comment]
Same here on k3s using https://github.com/kubeflow/manifests/blob/master/kfdef/kfctl_k8s_istio.v1.1.0.yaml

```bash
k3s --version
k3s version v1.18.8+k3s1 (6b595318)
```

```bash
...
servicerolebinding.rbac.istio.io/bind-cache-server-admission-webhook unchanged
servicerolebinding.rbac.istio.io/bind-gateway-ml-pipeline-ui unchanged
servicerolebinding.rbac.istio.io/bind-ml-pipeline-internal unchanged
persistentvolumeclaim/katib-mysql unchanged
persistentvolumeclaim/metadata-mysql unchanged
persistentvolumeclaim/minio-pvc unchanged
persistentvolumeclaim/mysql-pv-claim unchanged
validatingwebhookconfiguration.admissionregistration.k8s.io/seldon-validating-webhook-configuration-kubeflow configured
ERRO[0728] Permanently failed applying application kubeflow-apps:  (kubeflow.error): Code 500 with message: Apply.Run : error when creating "/tmp/kout183651406": CustomResourceDefinition.apiextensions.k8s.io "seldondeployments.machinelearning.seldon.io" is invalid: [spec.validation.openAPIV3Schema.properties[spec].properties[predictors].items.properties[componentSpecs].items.properties[spec].properties[containers].items.properties[ports].items.properties[protocol].default: Required value: this property is in x-kubernetes-list-map-keys, so it must have a default or be a required property, spec.validation.openAPIV3Schema.properties[spec].properties[predictors].items.properties[componentSpecs].items.properties[spec].properties[initContainers].items.properties[ports].items.properties[protocol].default: Required value: this property is in x-kubernetes-list-map-keys, so it must have a default or be a required property, spec.validation.openAPIV3Schema.properties[spec].properties[predictors].items.properties[explainer].properties[containerSpec].properties[ports].items.properties[protocol].default: Required value: this property is in x-kubernetes-list-map-keys, so it must have a default or be a required property]  filename="kustomize/kustomize.go:270"
Error: failed to apply:  (kubeflow.error): Code 500 with message: kfApp Apply failed for kustomize:  (kubeflow.error): Code 500 with message: Apply.Run : error when creating "/tmp/kout183651406": CustomResourceDefinition.apiextensions.k8s.io "seldondeployments.machinelearning.seldon.io" is invalid: [spec.validation.openAPIV3Schema.properties[spec].properties[predictors].items.properties[componentSpecs].items.properties[spec].properties[containers].items.properties[ports].items.properties[protocol].default: Required value: this property is in x-kubernetes-list-map-keys, so it must have a default or be a required property, spec.validation.openAPIV3Schema.properties[spec].properties[predictors].items.properties[componentSpecs].items.properties[spec].properties[initContainers].items.properties[ports].items.properties[protocol].default: Required value: this property is in x-kubernetes-list-map-keys, so it must have a default or be a required property, spec.validation.openAPIV3Schema.properties[spec].properties[predictors].items.properties[explainer].properties[containerSpec].properties[ports].items.properties[protocol].default: Required value: this property is in x-kubernetes-list-map-keys, so it must have a default or be a required property]
```

**Comment by DavidSpek** [comment]
I'm currently attempting to deploy kubeflow 1.1.0 using kfctl on a charmed kubernetes 1.19 cluster and seem to have the same issue. I am still working out how to enable istio support in the cluster so I'm not sure if that is contributing to this issue or not. 

```
kfctl exited with error: failed to apply:  (kubeflow.error): Code 500 with message: kfApp Apply failed for kustomize:  (kubeflow.error): Code 500 with message: Apply.Run : error when creating "/tmp/kout038140182": CustomResourceDefinition.apiextensions.k8s.io "seldondeployments.machinelearning.seldon.io" is invalid: [spec.validation.openAPIV3Schema.properties[spec].properties[predictors].items.properties[componentSpecs].items.properties[spec].properties[containers].items.properties[ports].items.properties[protocol].default: Required value: this property is in x-kubernetes-list-map-keys, so it must have a default or be a required property, spec.validation.openAPIV3Schema.properties[spec].properties[predictors].items.properties[componentSpecs].items.properties[spec].properties[initContainers].items.properties[ports].items.properties[protocol].default: Required value: this property is in x-kubernetes-list-map-keys, so it must have a default or be a required property, spec.validation.openAPIV3Schema.properties[spec].properties[predictors].items.properties[explainer].properties[containerSpec].properties[ports].items.properties[protocol].default: Required value: this property is in x-kubernetes-list-map-keys, so it must have a default or be a required property]
```

**Comment by DavidSpek** [comment]
I managed to get Charmed Kubernetes configured in a way that it supports istio with kfctl. The somewhat solution can be seen in the second part of my comment [here](https://github.com/juju-solutions/bundle-kubeflow/issues/240#issuecomment-692758081).

**Comment by DavidSpek** [comment]
Related: https://github.com/kubeflow/kfserving/issues/1070

**Comment by DavidSpek** [comment]
This has been resolved in Kubeflow 1.2 as it uses an updated version of Seldon.
